syft/internal/cache/dir.go

```diff
@@ -0,0 +1,71 @@
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/adrg/xdg"
+	"github.com/mitchellh/go-homedir"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+// SetRootDir sets the root directory to use for downloaded content caching for things like module license information
+func SetRootDir(dir string) {
+	rootDir = dir
+}
+
+// rootDir if set, will be the default root directory used to cache content
+var rootDir = ""
+
+// findRoot returns a directory to be used for caching, first available in this order:
+//   - rootDir
+//   - XDG_CACHE_HOME
+//   - HOME/.cache
+//   - TEMP
+func findRoot() string {
+	if rootDir != "" {
+		dir, err := homedir.Expand(rootDir)
+		if err != nil {
+			log.Warnf("unable to expand cache directory %s: %v", rootDir, err)
+		} else {
+			return dir
+		}
+	}
+
+	var err error
+	cacheRoot := xdg.CacheHome
+	if cacheRoot == "" {
+		cacheRoot, err = homedir.Dir()
+		if err != nil {
+			cacheRoot = os.TempDir()
+			log.Debugf("unable to get home directory due to: %v, defaulting cache to temp dir: %s", err, cacheRoot)
+		} else {
+			cacheRoot = filepath.Join(cacheRoot, ".cache")
+		}
+	}
+
+	return filepath.Join(cacheRoot, "syft")
```

Comment:
I feel like this could be done in `init()` and set as the initial `rootDir` value, thus we are really always using the static `rootDir` value (either set by the user or determined at `init`).

Reply:
This is no longer relevant with the latest changes: the configuration sets the dir once and passes a configured manager.